typer/main.py

```diff
@@ -816,8 +822,9 @@ def get_click_param(
     is_tuple = False
     parameter_type: Any = None
     is_flag = None
+    is_union_type = lenient_issubclass(type(main_type), UnionType)
```

Comment:
Hi, I'm new to this kind of stuff, could you please clarify this for me please? From the variable name, it is a `type` already, why do we need to convert it to `type()` here? Is it just your preference or there are some edge cases being handled implicitly here?